### PR TITLE
implement user group in the userprofil

### PR DESCRIPTION
**Pull request by sirinebettaieb**
Implement userGroup component in the profile page 
create groups and  that allow editing of groups that are displayed as avatars and name of groups

### CODE REVIEW

**Review by MoubarkiS — APPROVED** [review]

File: src/app/components/create-group-dialog/create-group-dialog.component.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
+<h1 mat-dialog-title>Créer un groupe</h1>
```
Comment (Benja-san) [inline src/app/components/create-group-dialog/create-group-dialog.component.html]: if it is a component, then it is injected inside a page so it won't be a H1 but h2 or higher number

File: src/app/components/edit-group-dialog/edit-group-dialog.component.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<h1 mat-dialog-title>Éditer le groupe</h1>
```
Comment (Benja-san) [inline src/app/components/edit-group-dialog/edit-group-dialog.component.html]: same here

File: src/app/components/user-groupe/user-groupe.component.html [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<div class="container">
+    <h1>Groupes</h1>
```
Comment (Benja-san) [inline src/app/components/user-groupe/user-groupe.component.html]: same here

File: src/app/model/groupes/groupe.ts
Comment (Benja-san) [inline src/app/model/groupes/groupe.ts:1]: Create a model for each one of those

File: src/app/service/group.service.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    group.members = group.members.map(member => ({...member,avatarUrl: 'assets/pictures/photo.png'}));
+    this.groups.push(group);
+    this.groupsSubject.next(this.groups);
+    return new BehaviorSubject<Group>(group).asObservable();
+  }
+
+  
+  deleteGroup(index: number): Observable<void> {
+    this.groups.splice(index, 1);
+    return of();
```
Comment (Benja-san) [inline src/app/service/group.service.ts]: you don't have to return an observable if you just delete a User

**Review by Benja-san — CHANGES_REQUESTED** [review]

File: src/app/model/groupes/groupe.ts
Comment (Benja-san) [reply src/app/model/groupes/groupe.ts:1]: use class over interface

File: src/app/model/membres/membre.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+export interface Member {
```
Comment (Benja-san) [inline src/app/model/membres/membre.ts]: here too

**Review by Benja-san — CHANGES_REQUESTED** [review]

**Review by Benja-san — APPROVED** [review]

**Pull request merged**